app/controllers/application_controller.rb

```diff
@@ -1,10 +1,15 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :create_cart
```

Comment:
In this set up we are creating a cart as soon as a visitor opens the app. That allows us to have the cart info persisted even if the user chooses to leave the application. Next time he visits the site the cart will still be there.